src/pages/EntityDetails/App/App.test.js

```diff
@@ -115,4 +116,74 @@ describe("Entity Details App", () => {
     await simulateSelectAllChange(false);
     unitsListChecked(false);
   });
+
+  it("enable the action button row when a unit is selected", async () => {
+    const wrapper = await generateComponent();
+    const findActionButton = () =>
+      wrapper.find('button[data-test="run-action-button"]');
+    const findSelectedUnits = () => wrapper.find('input[name="selectedUnits"]');
+    expect(findActionButton().prop("disabled")).toBe(true);
+    const firstInput = findSelectedUnits().at(0);
+    firstInput.simulate("change", {
+      target: { name: "selectedUnits", value: firstInput.prop("value") },
+    });
+    await waitForComponentToPaint(wrapper);
+    expect(findActionButton().prop("disabled")).toBe(false);
+  });
+
+  it("updates the url when units are selected and deselected", async () => {
+    const store = mockStore(dataDump);
+    const history = createMemoryHistory({
+      initialEntries: [
+        "/models/user-island@external/canonical-kubernetes/app/etcd",
+      ],
+    });
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <TestRoute path="/models/:userName/:modelName?/app/:appName?">
+            <QueryParamProvider ReactRouterRoute={Route}>
+              <App />
+            </QueryParamProvider>
+          </TestRoute>
+        </Router>
+      </Provider>
+    );
+    await waitForComponentToPaint(wrapper);
+
+    const findActionButton = () =>
+      wrapper.find('button[data-test="run-action-button"]');
+    const findSelectedUnits = () => wrapper.find('input[name="selectedUnits"]');
+
+    // Select a single unit.
+    const firstInput = findSelectedUnits().at(0);
+    firstInput.simulate("change", {
+      target: { name: "selectedUnits", value: firstInput.prop("value") },
```

Comment:
If I do that then we will have to update this test when our data set changes. Doing it this way it sends the value of the input (like it would in the browser).

Reply:
My point is that in the assert you are hardcoding the expected value. So it either all should be hardcoded or dynamic to saving having to update the tests.